doc/man3/SSL_set_blocking_mode.pod

```diff
@@ -0,0 +1,69 @@
+=pod
+
+=head1 NAME
+
+SSL_set_blocking_mode, SSL_get_blocking_mode - configure blocking mode for a
+QUIC SSL object
+
+=head1 SYNOPSIS
+
+ #include <openssl/ssl.h>
+
+ __owur int SSL_set_blocking_mode(SSL *s, int blocking);
+ __owur int SSL_get_blocking_mode(SSL *s);
+
+=head1 DESCRIPTION
+
+SSL_set_blocking_mode() can be used to enable or disable blocking mode on a QUIC
+connection SSL object. By default, blocking is enabled, unless the SSL object is
+configured to use an underlying read or write BIO which cannot provide a poll
+descriptor (see L<BIO_get_rpoll_descriptor(3)>), as blocking mode cannot be
+supported in this case.
+
+To enable blocking mode, call SSL_set_blocking_mode() with B<blocking> set to 1;
+to disable it, call SSL_set_blocking_mode() with B<blocking> set to 0.
+
+To retrieve the current blocking mode, call SSL_get_blocking_mode().
+
+Blocking mode means that calls such as SSL_read() and SSL_write() will block
+until the requested operation can be performed. In nonblocking mode, these
+calls will fail if the requested operation cannot be performed immediately; see
+L<SSL_get_error(3)>.
+
+These functions are only applicable to QUIC connection SSL objects. Other kinds
+of SSL object, such as those for TLS, automatically function in blocking or
```

Comment:
Hmm... how? libssl TLS is basically written assuming non-blocking I/O and if you feed it a blocking socket it just "happens" to never block from libssl's perspective, so none of the EWOULDBLOCK etc. handling code comes into play.

If an application passes a blocking BIO to a TLS SSL object and this function is called to set non-blocking mode, what are we supposed to do? Conversely if an application passes a non-blocking BIO to a TLS SSL object and this function is called to set blocking mode, what then?

Reply:
Well I was mainly thinking about the case where we have a non-blocking BIO and this function is called to set blocking mode (I'm not sure the other way around is interesting since we don't support that in QUIC either). But if we have a non-blocking BIO then why can't we do the same thing as you're doing in this PR for QUIC? i.e. sit in a `poll()` loop waiting for something interesting to happen on the poll fds. This of course means the app can't use its own event loop in that scenario - but the same is true for QUIC.

Reply:
In principle we could support blocking I/O on TLS/DTLS in this way as well. Not really for this PR though and something to be discussed.